src/group_theory/commensurable.lean

```diff
@@ -0,0 +1,145 @@
+import group_theory.index
+import group_theory.quotient_group
+import tactic.linarith
+import group_theory.subgroup.pointwise
+
+namespace commensurable
+
+open_locale pointwise
+
+variables {G : Type*} [group G]
+
+def commensurable (H K : subgroup G) : Prop :=
+  subgroup.index (subgroup.subgroup_of H K) ≠ 0 ∧
+  subgroup.index (subgroup.subgroup_of K H) ≠ 0
+
+lemma inf_self_eq_self (H :subgroup G)  : (H ⊓ H).comap H.subtype = ⊤ :=
+begin
+simp only [inf_idem],
+ext,
+simp only [set_like.coe_mem, subgroup.coe_subtype, subgroup.mem_top, iff_self, subgroup.mem_comap],
+end
+
+lemma subgroup_of_top (H : subgroup G) : subgroup.subgroup_of H H = ⊤ :=
+begin
+rw subgroup.subgroup_of,
+have:= inf_self_eq_self H,
+simp at *,
+exact this,
+end
```

Comment:
Golfed:
```lean
lemma comap_self_subtype (H :subgroup G) : H.comap H.subtype = ⊤ :=
eq_top_iff.2 $ λ x _, x.prop

lemma inf_self_eq_self (H :subgroup G)  : (H ⊓ H).comap H.subtype = ⊤ :=
by rw [inf_idem, comap_self_subtype]

lemma subgroup_of_top (H : subgroup G) : subgroup.subgroup_of H H = ⊤ :=
comap_self_subtype H
```

Reply:
oh awesome, thanks! I dont actually know what I'll need yet, I'm trying to prove the transitivity of commensurable and give the definition of the commensurator subgroup. From there I need to check what else I want for defining Hecke algebras.  This is a long way of saying, dont worry about golfing this just yet as I don't know what I'll include in the final PR, but that said, I very much appreciate the help!